src/main/java/com/google/devtools/build/lib/remote/http/HttpCacheClient.java

```diff
@@ -285,64 +287,58 @@ public void channelCreated(Channel ch) {
   }
 
   @SuppressWarnings("FutureReturnValueIgnored")
-  private Channel acquireUploadChannel() throws InterruptedException {
+  private Promise<Channel> acquireUploadChannel() {
     Promise<Channel> channelReady = eventLoop.next().newPromise();
     channelPool
-        .acquire()
-        .addListener(
-            (Future<Channel> channelAcquired) -> {
-              if (!channelAcquired.isSuccess()) {
-                channelReady.setFailure(channelAcquired.cause());
-                return;
-              }
-
-              try {
-                Channel ch = channelAcquired.getNow();
-                ChannelPipeline p = ch.pipeline();
-
-                if (!isChannelPipelineEmpty(p)) {
-                  channelReady.setFailure(
-                      new IllegalStateException("Channel pipeline is not empty."));
-                  return;
-                }
-
-                p.addFirst(
-                    "timeout-handler",
-                    new IdleTimeoutHandler(timeoutSeconds, WriteTimeoutException.INSTANCE));
-                p.addLast(new HttpResponseDecoder());
-                // The 10KiB limit was chosen arbitrarily. We only expect HTTP servers to respond
-                // with an error message in the body, and that should always be less than 10KiB. If
-                // the response is larger than 10KiB, HttpUploadHandler will catch the
-                // TooLongFrameException that HttpObjectAggregator throws and convert it to an
-                // IOException.
-                p.addLast(new HttpObjectAggregator(10 * 1024));
-                p.addLast(new HttpRequestEncoder());
-                p.addLast(new ChunkedWriteHandler());
-                synchronized (credentialsLock) {
-                  p.addLast(new HttpUploadHandler(creds, extraHttpHeaders));
-                }
+            .acquire()
+            .addListener(
+                    (Future<Channel> channelAcquired) -> {
+                      if (!channelAcquired.isSuccess()) {
+                        channelReady.setFailure(channelAcquired.cause());
+                        return;
+                      }
+
+                      try {
+                        Channel ch = channelAcquired.getNow();
+                        ChannelPipeline p = ch.pipeline();
```

Comment:
nit: can you please change `p` to `pipeline` (or similar)?

Reply:
Fixed. I changed the download side as well.